src/corva/models/scheduled.py

```diff
@@ -48,6 +54,23 @@ class ScheduledEvent(BaseEvent):
         None, description='Day shift start time'
     )
 
+    @pydantic.validator('schedule_start', 'schedule_end')
+    def set_timestamp(cls, v) -> int:
+        """Casts Unix timestamp from milliseconds to seconds.
+
+        Casts Unix timestamp from millisecond to seconds, if provided timestamp is in
+          milliseconds.
+        """
+
+        try:
+            # the function will raise, if provided timestamp is in milliseconds,
+            # because the year will always be too big to fit into the datetime instance
+            datetime.utcfromtimestamp(v)
+        except ValueError:
```

Comment:
There's no guarantee that the data is going to be an integer in milliseconds if it throws a ValueError, is there? I believe this will throw an OverflowError. It may also be better to just check if the timestamp is > the max possible timestamp.

Reply:
Good idea, reworked the logic